### ISSUE TITLE
Runtime error trying open SMILEI HDF5 

### ISSUE THREAD

**Issue by kevinCassou** [openPMD/openPMD-api#1238]
Dear openPMD developers, 

We are trying to shift to `openPMD-viewer` for post-processing of [SMILEI ](https://smileipic.github.io/Smilei/) PIC code data output. 
Following the doc we tried  
```
series = io.Series(path,io.Access.read_only)
```
and we got :
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-53-76e6870bcf32> in <module>
----> 1 series = io.Series(path,io.Access.read_only)

RuntimeError: [setAttribute] Value for string attribute 'name' must not be empty!
```
It looks from the **openPMD** standard technical files that it should not be a problem. Main differences (from documentation based on WARP )  that SMILEI PIC code generates one HDF5 for fields and one for tracked particles  with all the timesteps inside;

Then looking to the HDF5 structure of PIC data **SMILEI** :   `TrackParticles_species.h5` and `Fields0.h5`. The structure seems to respect openPMD standard. 

* For particles *data/timestep/particles/species/....* . In the metadata we have the `name=`attribute is empty : 
```
/ (96)
    Group size = 1
    Number of attributes = 11
        basePath = /data/%T/
        date = 2022-02-12 20:32:23 +�
        iterationEncoding = groupBased
        iterationFormat = /data/%T/
        meshesPath = no_meshes
        name = 
        openPMD = 1.0.0
        openPMDextension = 0
        particlesPath = particles/
        software = Smilei
        softwareVersion = ??-??
```

* For the fields it looks less similar as there is not sub repertory *fields* but again  data/timestep/El_mode_0 ...* . In the metadata we have also  `name=`attribute empty : 
```
/ (96)
    Group size = 2
    Number of attributes = 11
        basePath = /data/%T/
        date = 2022-02-12 20:32:23 +�
        iterationEncoding = groupBased
        iterationFormat = /data/%T/
        meshesPath = 
        name = 
        openPMD = 1.0.0
        openPMDextension = 0
        particlesPath = no_particles
        software = Smilei
        softwareVersion = ??-??
```
Some minor adjustement might be required on SMILEI could you guide us on it ? or is that a bug with empty attributes on `openpmd-api` ? 

**Software Environment:**
 - version of openPMD-api: `openpmd-api               0.14.4          nompi_py37hf1a3fc2_101    conda-forge`
 - installed openPMD-api via: [conda-forge]

**Comment by franzpoeschel** [comment]
Hello,
this issue is known and has been [recently fixed](https://github.com/openPMD/openPMD-api/pull/1223). If you want to try the dev branch, then I expect things should work.

Background: We don't support writing empty strings due to [corner cases](https://github.com/openPMD/openPMD-api/issues/1221#issuecomment-1066182184) in HDF5. Checking for empty strings accidentally leaked into reading due to shared code.

**Comment by ax3l** [comment]
Fixed in the `0.14.5` release.
Thank you for the detailed report and please let us know if you encounter any other errors in the future @kevinCassou :+1:
